ceph/ceph_admin/helper.py

```diff
@@ -185,7 +185,7 @@ def generate_host_spec(self, spec):
             if labels:
                 # Skipping client node, if only client label is attached
                 if len(node.role.role_list) == 1 and ["client"] == node.role.role_list:
-                    return
+                    continue
```

Comment:
would there be template render exception, suppose if we have one spec content for host with only one node with client label?
```
          - config:
              command: apply_spec
              service: orch
              specs:
                - service_type: host
                  address: true
                  labels: apply-all-labels
                  nodes:
                    - node4
```

Reply:
can we take away client node if we do not want to use it in cluster and continue with test suite scenario.
```
          - config:
              command: apply_spec
              service: orch
              specs:
                - service_type: host
                  address: true
                  labels: apply-all-labels
                  nodes:
                    - node2
                - service_type: host
                  address: true
                  labels: apply-all-labels
                  nodes:
                    - node3
 ```

Reply:
Or you could another node if that make sense.###

Reply:
and rgw deployment step uses node4 again. I think its better we add one more node to the conf.

Reply:
Yes sunil RGW config was added to node4 so yea I think I will add another node, node5 as a client node. And for client nodes, if the role added in conf file is only client, according to the current code it skips adding that node to the cluster. So with this change I guess we should be good. 

tldr: We will be adding new node for client node5 and this will not be part of the cluster

Reply:
Changes done. Could you please review?

Reply:
@sunilkumarn417 I tested by adding the spec content like you mentioned and there was no template render exception. 

          - config:
              command: apply_spec
              service: orch
              specs:
                - service_type: host
                  address: true
                  labels: apply-all-labels
                  nodes:
                    - node5

https://159.23.92.24/job/rhceph-test-execution-manual/14/console
https://159.23.92.24/job/rhceph-test-execution-manual/15/console

So we can retain this spec or remove it, it wouldn't make a difference.